_studio/mfx_lib/encode/mjpeg/src/mfx_mjpeg_encode.cpp

```diff
@@ -1936,11 +1936,11 @@ mfxStatus MFXVideoENCODEMJPEG::EncodeFrameCheck(mfxEncodeCtrl *ctrl, mfxFrameSur
 
     if (MFX_ERR_NONE == sts || MFX_WRN_INCOMPATIBLE_VIDEO_PARAM == sts || MFX_ERR_MORE_BITSTREAM == sts)
     {
+        // EncodeFrameCheck(ctrl, surface, bs, reordered_surface, pInternalParams) processes
+        // !surface case, so here we must have pOriginalSurface.
+        assert(pOriginalSurface);
```

Comment:
It can't be dereferenced. I added the comment that the check against nullptr is done above, in the mentioned function. `assert` is for a hypothetical situation like "the check was dropped from where it was, and in debug build we got an assert for some test". 
`assert` can be dropped if you see no value in it.

Reply:
Better drop please. And change "we must have pOriginalSurface" to smth like "pOriginalSurface isn't nullptr"

Reply:
> this assertion is useless. 

In bug reports it's more convenient to see "assertion triggered at ..." then just "test fail to segfault". So I tend to disagree.

> Why you decided to add assertion here? 

I described that in https://github.com/Intel-Media-SDK/MediaSDK/pull/1078#discussion_r246781096 . Sure the reason in not that significant.

Reply:
- pOriginalSurface isn't 0 here. Right? Don't care about ghosts.
- Bug reports are from release builds where assert is empty. So only segfault, but never expected.

Reply:
Ahh, developers and only developers run debug builds, missed that.

Reply:
So, the assertion was dropped.